colmena/proxy.py

```diff
@@ -1,18 +1,65 @@
 """Utilities for interacting with ProxyStore"""
 import logging
 import warnings
-import proxystore as ps
 
-from typing import Any, Union, List
+import proxystore
+from proxystore.store.endpoint import EndpointStore
+from proxystore.store.file import FileStore
+from proxystore.store.globus import GlobusStore
+from proxystore.store.local import LocalStore
+from proxystore.store.redis import RedisStore
+from proxystore.proxy import extract
+from proxystore.proxy import is_resolved
+from proxystore.proxy import Proxy
+from proxystore.store.base import Store
+from proxystore.store.utils import resolve_async
+
+from typing import Any, Union, List, Optional, Type
 
 logger = logging.getLogger(__name__)
 
+STORES = {
+    'EndpointStore': EndpointStore,
+    'FileStore': FileStore,
+    'GlobusStore': GlobusStore,
+    'LocalStore': LocalStore,
+    'RedisStore': RedisStore,
+}
+
 
 class ProxyJSONSerializationWarning(Warning):
     pass
 
 
-def proxy_json_encoder(proxy: ps.proxy.Proxy) -> Any:
+def get_store(
+    name: str,
+    kind: Optional[Union[Type[Store], str]] = None,
+    **kwargs: Any,
+) -> Optional[Store]:
+    """Get a Store by name or create one if none exists.
+
+    Args:
+        name (str): name of the store.
+        kind (type[Store], str): type of store to initialize if one with the
+            same `name` is not found. If a string, the correct type will be
+            resolved using the ``STORES`` mapping. If ``None`` (the default),
+            no store will be initialized.
```

Comment:
What's the use case for a `kind` of `None`? Is that when you know the store is already initialized?

Reply:
Would you mind including the explanation of when `kind` can be None in the documentation?

Reply:
Will do. Originally I had it as not optional, but it looked strange in testing to pass the type of `Store` to initialize when the store is already initialized. It would be useful if you knew the name of the `Store` but not the type (though that use does not exist in the current implementation).